tests/virt_autotest/validate_host_and_guests.pm

```diff
@@ -0,0 +1,48 @@
+# SUSE's openQA tests
+#
+# Copyright 2012-2022 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+#
+# Summary: Do a basic examination to the host and guests after tests run
+# Maintainer: qe-virt@suse.de
+
+package validate_host_and_guests;
+
+use strict;
+use warnings;
+use base "virt_feature_test_base";
+use testapi;
+use virt_autotest::common;
+use virt_autotest::utils qw(check_host_health check_guest_health);
+
+sub run_test {
+
+    my $failures = check_host_health;
+    die "Host is not healthy: $failures" unless $failures eq "";
+    foreach my $guest (keys %virt_autotest::common::guests) {
+        $failures = check_guest_health($guest);
+        die "Guest $guest is not healthy: $failures" unless $failures eq "";
+    }
+}
+
+sub prepare_run_test {
+    script_run("rm -f /root/{commands_history,commands_failure}");
+    assert_script_run("history -c");
+}
+
+sub post_fail_hook {
+    my $self = shift;
+    collect_virt_system_logs();
```

Comment:
You do not need this ```collect_virt_system_logs```, because ```virt_utils::collect_host_and_guest_logs``` collects everything. ```collect_virt_system_logs``` was just developed by MU years ago and being used by them.